### PR TITLE
fix(scroll): 修复滚动表格后, 鼠标移动会重置已选中单元格 close #904

### LINKED ISSUE #904: 🐛滚动表格后，鼠标移动会重置已选的单元格
### 🏷 Version

| Package      | Version |
| -------------- | --------- |
| @antv/s2       | -1.3.0         |
| @antv/s2-react | - 1.3.0        |

### 🖋 Description

滚动表格后，鼠标移动会激活高亮十字动作，已选的单元格消失（重置）
### ⌨️ Code Snapshots

例子：https://s2.antv.vision/zh/examples/case/kpi-strategy#basic

### 🤔 Steps to Reproduce
单选/多选单元格，滚动表格（需要滚动条），移动鼠标，高亮消失

### 😊 Expected Behavior
已选的单元格高亮维持
### 😅 Current Behavior
已选单元格消失
### 💻 System information
chrome: v96 Macos

### PR DESCRIPTION
**Pull request by lijinke666**
### 👀 PR includes

✨ Feature

- [ ] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [x] Solve the issue and close #904 

🔧 Chore

- [ ] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 📝 Description

RT

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| Kapture 2021-12-18 at 15 03 02| Kapture 2021-12-18 at 15 01 18|

### 🔍 Self Check before Merge

- [x] Add or update relevant Docs.
- [x] Add or update relevant Demos.
- [x] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-core/src/facet/base-facet.ts [hunk 1]
```diff
@@ -1151,7 +1151,11 @@ export abstract class BaseFacet {
   }
 
   protected onAfterScroll = debounce(() => {
-    this.spreadsheet.interaction.removeIntercepts([InterceptType.HOVER]);
+    const { interaction } = this.spreadsheet;
+    // 如果是选中单元格状态, 则不取消 hover
```
Comment (xingwanying) [inline packages/s2-core/src/facet/base-facet.ts:1155]: 为啥不是选中状态才取消 hover ？

Comment (lijinke666) [reply packages/s2-core/src/facet/base-facet.ts:1155]: 选中单元格后会添加 hover 拦截
而对于滚动, 开始滚动: 添加hover拦截, 滚动结束: 移除hover拦截, 所以选中且滚动后, hover拦截被移除了, 这时候继续hover, 则选中的单元格被重置了, 所以这里这样处理呀

**Review by xingwanying — APPROVED** [review]
LG

**Pull request merged**